Models/Terminal.php

```diff
@@ -44,10 +46,10 @@ class Altapay_Models_Terminal extends ObjectModel
      *
      * @throws PrestaShopDatabaseException
      */
-    public static function getTerminals()
+    public static function getTerminals($shop_id = 1)
     {
         return Db::getInstance(_PS_USE_SQL_SLAVE_)->ExecuteS('
-			SELECT * FROM `' . _DB_PREFIX_ . 'altapay_terminals` ORDER BY `id_terminal` ASC
+			SELECT * FROM `' . _DB_PREFIX_ . 'altapay_terminals` WHERE shop_id = ' . (int) $shop_id . ' ORDER BY `id_terminal` ASC
```

Comment:
If the shop_id has characters other than numbers, then casting it to `int` will return `0`, which in itself can be a real `shop_id`, returning results that are not desired.

Please perform proper validation and throw validation error in case the shop_id is not correct.

Reply:
@emicha, I have checked the type of shop_id in the shop table it is defined as int. Also, that's how Prestashop uses shop_id in their core files. 
<img width="765" alt="image" src="https://user-images.githubusercontent.com/15101585/213400023-1054d4a2-8d8a-4d8e-a54b-0b9828a5113d.png">

Reply:
@BushraAsif The method is public
```php
public static function getTerminals($shop_id = 1)
```
This means that in the future, the shop_id will come from other sources that might not sanitised properly.

If the shop_id has the value `1asd` for example, casting it will force it to be 0, which will create side effects.

Reply:
Updated